src/implementations/orthnull.jl

```diff
@@ -79,78 +79,75 @@ function initialize_output(::typeof(right_null!), A::AbstractMatrix)
     return Nᴴ
 end
 
+function algorithm_or_select_algorithm(f, A::AbstractMatrix, alg::AbstractAlgorithm)
+    return alg
+end
+function algorithm_or_select_algorithm(f, A::AbstractMatrix, kwargs::NamedTuple)
+    return select_algorithm(f, A; kwargs...)
+end
+
```

Comment:
Perhaps this definition makes slightly more sense to migrate to the `src/algorithms.jl` file?

Reply:
Yes that's a good point.

Reply:
I moved it and renamed `algorithm_or_select_algorithm` to `_select_algorithm`, which matches the name I used for similar functionality in #23, which I'm happy to change (though maybe the name and design can be discussed in #23 since the name is less important in this PR).